libraries/AP_Landing/AP_Landing.h

```diff
@@ -184,5 +204,57 @@ class AP_Landing
     bool type_slope_is_flaring(void) const;
     bool type_slope_is_on_approach(void) const;
     bool type_slope_is_expecting_impact(void) const;
-
+    bool type_slope_is_throttle_suppressed(void) const;
+
+    // Landing type TYPE_DEEPSTALL
+
+    //public AP_Landing interface
+    void type_deepstall_do_land(const AP_Mission::Mission_Command& cmd, const float relative_altitude);
+    void type_deepstall_verify_abort_landing(const Location &prev_WP_loc, Location &next_WP_loc, bool &throttle_suppressed);
+    bool type_deepstall_verify_land(const Location &prev_WP_loc, Location &next_WP_loc, const Location &current_loc,
+            const float height, const float sink_rate, const float wp_proportion, const uint32_t last_flying_ms,
+            const bool is_armed, const bool is_flying, const bool rangefinder_state_in_range);
+    void type_deepstall_setup_landing_glide_slope(const Location &prev_WP_loc, const Location &next_WP_loc,
+            const Location &current_loc, int32_t &target_altitude_offset_cm);
+    bool type_deepstall_override_servos(void);
+    bool type_deepstall_request_go_around(void);
+    bool type_deepstall_get_target_altitude_location(Location &location);
+    int32_t type_deepstall_get_target_airspeed_cm(void) const;
+    bool type_deepstall_is_throttle_suppressed(void) const;
+
+    const DataFlash_Class::PID_Info& type_deepstall_get_pid_info(void) const;
+
+    //private helpers
+    void type_deepstall_build_approach_path();
+    float type_deepstall_predict_travel_distance(const Vector3f wind, const float height) const;
+    bool type_deepstall_verify_breakout(const Location &current_loc, const Location &target_loc, const float height_error) const;
+    float type_deepstall_update_steering(void);
+
+    // deepstall members
+    enum deepstall_stage {
+        DEEPSTALL_STAGE_FLY_TO_LANDING,    // fly to the deepstall landing point
+        DEEPSTALL_STAGE_ESTIMATE_WIND,     // loiter until we have a decent estimate of the wind for the target altitude
+        DEEPSTALL_STAGE_WAIT_FOR_BREAKOUT, // wait until the aircraft is aligned for the optimal breakout
+        DEEPSTALL_STAGE_FLY_TO_ARC,        // fly to the start of the arc
+        DEEPSTALL_STAGE_ARC,               // fly the arc
+        DEEPSTALL_STAGE_APPROACH,          // fly the approach in, and prepare to deepstall when close 
+        DEEPSTALL_STAGE_LAND,              // the aircraft will stall torwards the ground while targeting a given point
+    };
+    deepstall_stage type_deepstall_stage;
+    Location type_deepstall_landing_point;
+    Location type_deepstall_extended_approach;
+    Location type_deepstall_breakout_location;
+    Location type_deepstall_arc;
+    Location type_deepstall_arc_entry;
```

Comment:
Could recompute them, but seemed more expensive then just storing the values once. Particularly because some of the later ones are several manipulations in. I could be convinced to recompute them as needed, but it would complicate the structure as we would be recomputing it in several functions that get independently called. Unless we absolutely need the flash space I think saving the computation time is probably the better approach as its overall cleaner IMO.